### PR TITLE
add typedoc [#287]

### PR DESCRIPTION
**Pull request by bdon**

### CODE REVIEW

File: js/index.ts [hunk 1]
```diff
@@ -834,6 +837,8 @@ export class SharedPromiseCache {
   }
 }
 
+// if source is a string, create a browser FetchSource().
```
Comment (HarelM) [inline js/index.ts:840]: This should be using /** type of comments to show in the docs as far as I know.

Comment (HarelM) [reply js/index.ts:840]: Might be worth adding this on the conductor method too, this way you would see it in vscode when you write the code to use the constructor.

**Comment by HarelM** [comment]
BTW, what are you doing with the generated docs? Are you publishing then to github pages?
If not, you might not need anything soecial as when you create the .d.ts file the comments with be there, and the IDE will pick it up.
So typedoc might be an overkill here.
IDK... ping me on slack if you want to pick my brain.

**Comment by bdon** [comment]
Not 100% sure if types are even working right now - can you confirm with 3.0.0-alpha.2? https://publint.dev/pmtiles@3.0.0-alpha.2

I plan to add the docs to GitHub pages, seems helpful for google-ability as well as for those not using IDEs

**Comment by HarelM** [comment]
I don't see any comments here:
https://unpkg.com/browse/pmtiles@3.0.0-alpha.2/dist/index.d.ts
Also there's an empty index2.d.ts file, see here:
https://unpkg.com/browse/pmtiles@3.0.0-alpha.2/dist/

**Comment by bdon** [comment]
thanks, removed the index2 file. I haven't added any comments yet, hence the typedoc isn't published

**Comment by HarelM** [comment]
The d.ts file should contain comments straight from the typescript code, regardless of typedoc.

**Comment by HarelM** [comment]
But it requires "/**" type of comments, I think.

**Pull request merged**